bigquery_etl/cli/query.py

```diff
@@ -918,24 +941,78 @@ def _run_query(
         if "query" not in query_arguments:
             query_arguments = ["query"] + query_arguments
 
+        query_text = render_template(
+            query_file.name,
+            template_folder=str(query_file.parent),
+            templates_dir="",
+            format=False,
+            **addl_templates,
+        )
+
+        # create a session, setting default project and dataset
+        if billing_project is not None and billing_project != project_id:
+            default_project, default_dataset, _ = extract_from_query_path(query_file)
+
+            session_id = create_query_session(
+                session_project=billing_project,
+                default_project=project_id or default_project,
+                default_dataset=dataset_id or default_dataset,
+            )
+            query_arguments.append(f"--session_id={session_id}")
+
         # write rendered query to a temporary file;
         # query string cannot be passed directly to bq as SQL comments will be interpreted as CLI arguments
         with tempfile.NamedTemporaryFile(mode="w+") as query_stream:
```

Comment:
I think you could avoid the need for writing to disk (and the `write()` and `seek()` calls) by using [`io.StringIO`](https://docs.python.org/3.11/library/io.html#io.StringIO).
```suggestion
        with io.StringIO(query_text) as query_stream:
```

Reply:
`StringIO` doesn't seem to work with the subprocess call because a file descriptor is required: `io.UnsupportedOperation: fileno`

One way that would work is
```py
with subprocess.Popen(["bq"] + query_arguments, stdin=subprocess.PIPE) as proc:
    proc.communicate(query_text.encode(encoding="utf-8"))
```
Another way would be removing comments from the sql with some simple regex but I don't see these as strictly better in terms of readability or performance so I'll opt to leave it since it's unrelated to this pr.